src/storage/__tests__/GitlabTokenStorage.test.ts

```diff
@@ -0,0 +1,486 @@
+    ).toHaveProperty('projectId', 35102363);
+    expect(
+      await storageProvider.assignProjectId(),
+    ).toHaveProperty('groupId', 51634506);
+  });
+
+  it('should fetch branches as a simple list', async () => {
+    mockGetBranches.mockImplementationOnce(() => (
+      Promise.resolve(
+        [
+          { name: 'main' },
+          { name: 'development' },
+        ],
+      )
+    ));
+
+    expect(
+      await storageProvider.fetchBranches(),
+    ).toEqual(
+      ['main', 'development'],
+    );
+  });
+
+  it('should try to create a branch', async () => {
+    mockCreateBranch.mockImplementationOnce(() => (
+      Promise.resolve({
+        name: 'development',
+      })
+    ));
+    expect(await storageProvider.createBranch('development', 'main')).toBe(true);
+    expect(mockCreateBranch).toBeCalledWith(35102363, 'development', 'heads/main');
+  });
+
+  it('create a branch should return false when it is failed', async () => {
+    mockCreateBranch.mockImplementationOnce(() => (
+      Promise.resolve({
+      })
+    ));
+    expect(await storageProvider.createBranch('development', 'main')).toBe(false);
+    expect(mockCreateBranch).toBeCalledWith(35102363, 'development', 'heads/main');
+  });
+
+  it('canWrite should return true if use is a collaborator by GroupMember', async () => {
+    mockGetCurrentUser.mockImplementationOnce(() => (
+      Promise.resolve({
+        id: 11289475,
+        state: 'active',
+      })
+    ));
+    mockGetGroupMembers.mockImplementationOnce(() => (
+      Promise.resolve({
+        access_level: 50,
+      })
+    ));
+    expect(await storageProvider.canWrite()).toBe(true);
+    expect(mockGetGroupMembers).toBeCalledWith(51634506, 11289475);
+  });
+
+  it('canWrite should return true if user is a collaborator by projectMember', async () => {
+    mockGetCurrentUser.mockImplementationOnce(() => (
+      Promise.resolve({
+        id: 11289475,
+        state: 'active',
+      })
+    ));
+    mockGetGroupMembers.mockImplementationOnce(() => (
+      Promise.reject(new Error())
+    ));
+    mockGetProjectMembers.mockImplementationOnce(() => (
+      Promise.resolve({
+        access_level: 50,
+      })
+    ));
+    expect(await storageProvider.canWrite()).toBe(true);
+    expect(mockGetProjectMembers).toBeCalledWith(35102363, 11289475);
+  });
+
+  it('canWrite should return false if user is not a collaborator', async () => {
+    mockGetCurrentUser.mockImplementationOnce(() => (
+      Promise.resolve({
+        id: 11289475,
+        state: 'active',
+      })
+    ));
+    mockGetGroupMembers.mockImplementationOnce(() => (
+      Promise.resolve({
+        access_level: 20,
+      })
+    ));
+    expect(await storageProvider.canWrite()).toBe(false);
+  });
+
+  it('can read from Git in single file format', async () => {
+    mockGetRepositories.mockImplementationOnce(() => (
+      Promise.resolve([])
+    ));
+
+    storageProvider.changePath('data/global.json');
```

Comment:
```suggestion
    storageProvider.changePath('data/tokens.json');
```

for this single file scenario it would be better to call this 'tokens.json' to remove confusion with the token set called 'global'